File: 프로그래머스/3/42898. 등굣길/등굣길.java
class Solution {
    public int solution(int m, int n, int[][] puddles) {
        int answer = 0;
        
        int[][] dp = new int[n+1][m+1];
        dp[1][1] = 1; // 시작 지점
        
        for (int[] p: puddles) {
            dp[p[1]][p[0]] = -1;
        }
        
        for (int i = 1; i < n+1; i++) {
            for (int j = 1; j < m+1; j++) {
                if (i == 1 && j == 1) {
                    continue;
                }
                if (dp[i][j] == -1) {
                    dp[i][j] = 0;
                } else {
                    dp[i][j] = (dp[i-1][j] + dp[i][j-1]) % 1000000007;
                }
            }
        }
        
        answer = dp[n][m];
        return answer;
    }
}

File: 프로그래머스/3/42898. 등굣길/등굣길.py
def solution(m, n, puddles):
    answer = 0
    
    dp = [[0] * m for _ in range(n)]
    dp[0][0] = 1 # 시작위치
    
    for i in range(n):
        for j in range(m):
            if i == 0 and j == 0:
                continue
            if [j+1, i+1] in puddles:
                dp[i][j] = 0
            else:
                dp[i][j] = (dp[i-1][j] + dp[i][j-1]) % 1000000007

    answer = dp[n-1][m-1]
    return answer